### PR TITLE
Mixtral MoE improvements: transposed w2 to have reduction dim be innermost dim

### PR DESCRIPTION
**Pull request by yanboliang**

### CODE REVIEW

**Review by Chillee — APPROVED** [review]

File: mixtral-moe/model.py [hunk 1]
```diff
@@ -183,21 +183,22 @@ def forward(self, x: Tensor, freqs_cis: Tensor, mask: Tensor, input_pos: Optiona
         y = self.wo(y)
         return y
 
+import torch.distributed
```
Comment (Chillee) [inline mixtral-moe/model.py:186]: remove this
